macro/src/main/scala/monocle/macros/GenIso.scala

```diff
@@ -59,27 +71,86 @@ class GenIsoImpl(val c: blackbox.Context) {
     """)
   }
 
-  def genIso_unit_impl[S: c.WeakTypeTag]: c.Expr[Iso[S, Unit]] = {
-    import c.universe._
-
+  protected def genIso_unit_tree[S: c.WeakTypeTag]: c.Tree = {
     val sTpe = weakTypeOf[S]
 
     if (sTpe.typeSymbol.isModuleClass) {
       val table = c.universe.asInstanceOf[SymbolTable]
       val tree = table.gen
-      val obj = tree.mkAttributedQualifier(sTpe.asInstanceOf[tree.global.Type]).asInstanceOf[c.universe.Tree]
-      c.Expr[Iso[S, Unit]](q"""
+      val obj = tree.mkAttributedQualifier(sTpe.asInstanceOf[tree.global.Type]).asInstanceOf[Tree]
+      q"""
         monocle.Iso[${sTpe}, Unit](Function.const(()))(Function.const(${obj}))
-      """)
+      """
     } else {
       caseAccessorsOf[S] match {
         case Nil =>
           val sTpeSym = sTpe.typeSymbol.companion
-          c.Expr[Iso[S, Unit]](q"""
+          q"""
             monocle.Iso[${sTpe}, Unit](Function.const(()))(Function.const(${sTpeSym}()))
-          """)
-        case _   => c.abort(c.enclosingPosition, s"$sTpe needs to be a case class with no accessor or an object.")
+          """
+        case _   => fail(s"$sTpe needs to be a case class with no accessor or an object.")
```

Comment:
Yeah, I find it helps readability in contrast to `c.abort(c.enclosingPosition, _)` all over. Hope you don't mind :)